### DISCUSSION TITLE
Invalid data searched in InputPicker

### DISCUSSION THREAD

**Discussion by PiousVenom** [rsuite/rsuite#2127] [Q&A]
Currently, if your search returns no results, if you hit enter, it triggers an onChange. If you click the "No results found" the input changes to the previously select until you focus on the input again, at which time it shows the search filter. And if you click elsewhere, it closes leaving the previously selected value.

What I'm wondering is if there's no results, I'd like to prevent the hitting enter to select, as well doing nothing if the user clicks the "No results found.". How can I go about doing that?

**Comment by SevenOutman** [comment]
Hi @PiousVenom, I agree that clicking "No results found" should result in no action. But I'm not sure what's the expected behavior upon hitting Enter. Should it do nothing or emit the current value of the input? WDYT?

/cc @simonguo

**Reply by PiousVenom** [reply]
I'd be inclined to say give an option to the developer. I can see where some would want it to reset, no action, or revert to last selected.

**Reply by simonguo** [reply]
> But I'm not sure what's the expected behavior upon hitting Enter. Should it do nothing or emit the current value of the input? WDYT?

@SevenOutman  In the absence of a result value, `Enter` triggers `onChange` This is a bug that needs to be fixed.  If `creatable = true `, then return the input value, otherwise do nothing.

> I'd be inclined to say give an option to the developer. I can see where some would want it to reset, no action, or revert to last selected.

@PiousVenom  Is [AutoComplete](https://rsuitejs.com/components/auto-complete/) more suitable for the application scenario you want?

**Reply by PiousVenom** [reply]
> @PiousVenom Is [AutoComplete](https://rsuitejs.com/components/auto-complete/) more suitable for the application scenario you want?

Not in this instance. I need a solution that offers a searchable dropdown.
